### PR TITLE
Indexing and Search with GTR Models in Pytorch

### PR DESCRIPTION
**Pull request by ToluClassics**

### CODE REVIEW

**Comment by ToluClassics** [comment]
PR based off https://github.com/lintool/fourdrinier/issues/636

File: pyserini/search/faiss/_searcher.py [hunk 1]
```diff
@@ -168,10 +168,11 @@ def _mean_pooling(model_output, attention_mask):
         sum_mask = torch.clamp(input_mask_expanded.sum(1), min=1e-9)
         return sum_embeddings / sum_mask
 
-    def encode(self, query: str):
+    def encode(self, query: str, max_length: int = 64):
```
Comment (lintool) [inline pyserini/search/faiss/_searcher.py:171]: is this change going to affect all our other models?

Comment (ToluClassics) [reply pyserini/search/faiss/_searcher.py:171]: No , it won't

**Comment by cramraj8** [comment]
Any reason why ? Does the GTR integration works with Pyserini ?

**Comment by cramraj8** [comment]
Any reason why GTR models were not integrated with Pyserini ? Are the current changes run without any errors and correctly reproduce the reported results?

**Comment by ToluClassics** [comment]
I'm not sure why we didn't end up merging this but I think the branch works, maybe you can pull that branch and try it out?

Alternatively is there a particular use case you need this for?

**Pull request closed without merging**